c++/src/kj/memory.h

```diff
@@ -150,6 +150,12 @@ class Own {
     return *this;
   }
 
+  template <typename... Attachments>
+  Own<T> attach(Attachments&&... attachments);
+  // Returns an Own<T> which points to the same object but which also ensures that all values
+  // passed to `attachments` remain alive until after this object is destroyed. Normally
+  // `attachments` are other Own<?>s pointing to objects that this one depends on.
```

Comment:
Maybe clarify that the order of destruction of `attachments` is the order in which they are passed? The comment in the implementation already makes this clear, but it'd be good to document at the interface definition, I think.